classes/04class/iac/README.md

```diff
@@ -0,0 +1,304 @@
+Just like any other programming language, Terraform has it's best practices to organize the code you create.
+
+Everything (resources, outputs, providers, state, etc) could be written in a single file with a `.tf` extension, however, to make things easier, each kind of code block should be created on a specific file.
+- **_provider.tf**
+  - All providers are declared on this file
+- **_backend.tf**
+  - The backend to be used is declared on this file
+- **variables.tf**
+  - The variables expected to be received during the code execution are declared on this file.
+- **_outputs.tf**
+  - All outputs from your code are declared on this file
+- **resource_related.tf**
+  - For the resources, you'll normally create a single file for each `family` of resources. Normally you'll have an `iam.tf`, `vpc.tf`, `s3.tf`, `rds.tf`, etc. In this example, all resources related to IAM, would go into the `iam.tf` file.
+- **anything.tfvars**
+  - In this file you would have the variables to be used during the code execution. If you have multiple environments, you can have your `dev.tfvars` and `prod.tfvars` files containing specific values for each environment and point to each file during execution based on the environment you want to apply your code.
+
+## Running some code
+Now that we have some understanding about terraform and its power let's give it a go and run some code.
+
+First of all, make sure you have your AWS credentials configured on your machine. You can use the guide from [here](https://link-to-the-cli-config)
+
+All the code we'll use is on the [artifacts folder](artifacts/terraform) of this repository, so just get into that directory. That way terraform will consider it as the working directory.
+
+```bash
+# Considering that you are on the folder of this file
+$ cd artifacts/terraform
+```
+Let's initialize terraform
+```bash
+$ terraform init
+```
+Now that terraform is initialized, we can run the next command, which would determine the resources that need to be created/updated/removed.
+```bash
+$ terraform plan -var-file=./class.tfvars
+```
+After this command, you should see 3 resources to be created in your account. Because it's the first time that terraform is running, it's basically creating all resources defined in the code.
+```bash
+$ terraform apply -var-file=./class.tfvars
+```
+By running the command above and saying `yes` to the question you'll receive, terraform will proceed with the creation of the resources informed.
+
+The resources should be created, and you can go into your AWS console and check that the resources listed below were successfully created:
+- S3 bucket: [8char]-devopsacademy
+- SecurityGroup: devopsacademy-sg
+
+Now edit the `class.tfvars` file and change the bucket name by adding a `-2` to its name.
+
+Additionally, edit the `sg.tf` file and change the port from 80 to 443 in the `from_port` and `to_port` arguments.
+
+Now let's re-run the following commands:
+```
+$ terraform init -var-file=./class.tfvars
```

Comment:
why the init again? maybe adding some description  on when do you need to use  init again

Reply:
fixed and added comment around the requirement to re-run init.